src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -114,6 +114,19 @@ FUZZ_TARGET(coinselection)
     if (total_balance >= target && subtract_fee_outputs && !HasErrorMsg(result_knapsack)) {
         assert(result_knapsack);
     }
+
+    std::vector<COutput> utxos;
+    std::set<std::shared_ptr<COutput>> utxos_set;
+    CreateCoins(fuzzed_data_provider, utxos, coin_params);
+    std::transform(utxos.begin(), utxos.end(),
+                 std::inserter(utxos_set, utxos_set.begin()),
+                 [](const COutput& output) {
+                   return std::make_shared<COutput>(output);
+                 });
+
+    if (result_srd) result_srd->AddInputs(utxos_set, subtract_fee_outputs);
+    if (result_knapsack) result_knapsack->AddInputs(utxos_set, subtract_fee_outputs);
+    if (result_bnb) result_bnb->AddInputs(utxos_set, subtract_fee_outputs);
```

Comment:
If `CreateCoins` generates a coin that is already inside any of the results, `AddInputs` will throw a runtime error, which the fuzzing framework will detect as a failure when it is not. It is part of the class boundaries.

Same apply for the fuzz `Merge` function commit.

Reply:
No ok. The reason why this doesn't fails is the test global increasing `next_locktime` which ensures that all UTXOs receive a different hash even when they have the same data.
So, nvm. A comment wouldn't hurt but nothing serious.

Reply:
Got it, thanks.